relay-event-schema/src/protocol/span.rs

```diff
@@ -177,6 +165,85 @@ impl Getter for Span {
     }
 }
 
+#[derive(Clone, Debug, Default, PartialEq, Empty, FromValue, IntoValue, ProcessValue)]
+#[cfg_attr(feature = "jsonschema", derive(JsonSchema))]
+pub struct SpanData {
+    /// The source code file name that identifies the code unit as uniquely as possible.
+    #[metastructure(field = "code.filepath", pii = "maybe")]
+    pub code_filepath: Annotated<Value>,
+    /// The line number in `code.filepath` best representing the operation.
+    #[metastructure(field = "code.lineno", pii = "maybe")]
+    pub code_lineno: Annotated<Value>,
+    /// The method or function name, or equivalent.
+    ///
+    /// Usually rightmost part of the code unit's name.
+    #[metastructure(field = "code.function", pii = "maybe")]
+    pub code_function: Annotated<Value>,
+    /// The "namespace" within which `code.function` is defined.
+    ///
+    /// Usually the qualified class or module name, such that
+    /// `code.namespace + some separator + code.function`
+    /// form a unique identifier for the code unit.
+    #[metastructure(field = "code.namespace", pii = "maybe")]
+    pub code_namespace: Annotated<Value>,
+
+    /// Other fields in `span.data`.
+    #[metastructure(additional_properties, pii = "true", retain = "true")]
+    other: Object<Value>,
+}
+
+impl SpanData {
+    /// Getter for compatibility with [`Object`].
+    pub fn get(&self, key: &str) -> Option<&Annotated<Value>> {
+        match key {
+            "code.filepath" => Some(&self.code_filepath),
+            "code.lineno" => Some(&self.code_lineno),
+            "code.function" => Some(&self.code_function),
+            "code.namespace" => Some(&self.code_namespace),
+            _ => self.other.get(key),
+        }
+    }
+
+    /// Setter for compatibility with [`Object`].
+    pub fn insert(&mut self, key: String, value: Annotated<Value>) {
```

Comment:
Do you think `set` could be more specific?

Reply:
This mimicks the interface of `Object = BTreeMap` (except it does not return the old value).